library/platform_util.c

```diff
@@ -72,7 +72,8 @@ static void * (* const volatile memset_func)( void *, int, size_t ) = memset;
 
 void mbedtls_platform_zeroize( void *buf, size_t len )
 {
-    memset_func( buf, 0, len );
+    if ( buf != NULL && len > 0 )
```

Comment:
Given the discussion in https://github.com/ARMmbed/mbed-crypto/pull/239#discussion_r320234363, we should probably change this check to just `if ( len > 0 )` and avoid the NULL check. This will allow static analyzers to see that `buf` is not allowed to be NULL for this function (and, as a very small bonus, we can save a tiny amount of code space).

Reply:
PSA and mbedtls APIs work differently here. In PSA Crypto, a null pointer dereference is supposed to invoke the platform's equivalent of a segmentation fault. In mbedtls, in low-level modules, a null pointer dereference is supposed to invoke `MBEDTLS_PARAM_FAILED` if `MBEDTLS_CHECK_PARAMS` is enabled (and segfaults otherwise), and many high-level modules (e.g. md, cipher and pk) return an error code on null pointers.

Here, I think the correct behavior is
```
MBEDTLS_INTERNAL_VALIDATE( len == 0 || buf != NULL );
if( len > 0 )
    memset_func( buf, 0, len );
```